src/model/interactions.rs

```diff
@@ -0,0 +1,109 @@
+//! Interactions information-related models
+
+use super::prelude::*;
+use crate::internal::prelude::*;
+
+/// Information about an Interaction
+///
+/// An interaction is the base "thing" that is sent when a user invokes a command, and is the same
+/// for Slash Commands and other future interaction types.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct Interaction {
+    pub id: InteractionId,
+    #[serde(rename = "type")]
+    pub interaction_type: InteractionType,
```

Comment:
`kind` is used as an alias for `type` across the models.
```suggestion
    pub kind: InteractionType,
```